sample-apps/go-sample-app/collection/client.go

```diff
@@ -59,59 +55,74 @@ var traceCommonLabels = []attribute.KeyValue{
 	attribute.String("port", cfg.Port),
 }
 
-// StartClient starts the OTEL controller which periodically collects signals and exports them.
+// StartClient starts the traces and metrics providers which periodically collects signals and exports them.
 // Trace exporter and Metric exporter are both configured.
 func StartClient(ctx context.Context) (func(context.Context) error, error) {
 
+	if id, present := os.LookupEnv("INSTANCE_ID"); present {
+		testingId = "_" + id
+	}
+	res := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		semconv.ServiceName("go-sample-app"),
+	)
+	if _, present := os.LookupEnv("OTEL_RESOURCE_ATTRIBUTES"); present {
+		envResource, err := resource.New(ctx, resource.WithFromEnv())
+		if err != nil {
+			return nil, err
+		} else {
```

Comment:
no need for else if you are going to return in the if.

Reply:
Remove the `else` in the latest commit